OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -437,11 +442,17 @@ public MovePart(Move move, WPos from, WPos to, WAngle fromFacing, WAngle toFacin
 			public override bool Tick(Actor self)
 			{
 				var mobile = Move.mobile;
-				moveFraction += mobile.MovementSpeedForCell(self, mobile.ToCell);
+
+				// If previous MovePart overshot MoveFractionTotal and we carried over a
```

Comment:
It took me a couple of goes to understand this. IMO a clearer way to present this would be to replace `needToMoveByStartingFraction` with a `firstTick` bool, and then change this to:
```csharp
// Having non-zero progress in the first tick means that this MovePart is following on from
// a previous MovePart that has just completed during the same tick. In this case, we want to
// apply the carried over progress but not evaluate a full new step.
if (!firstTick || progress == 0)
	progress += mobile.MovementSpeedForCell(self, mobile.ToCell);

firstTick = false;
```

What do you think?